src/banana/data/genpdf/load.py

```diff
@@ -0,0 +1,106 @@
+import pathlib
+import re
+
+import lhapdf
+import numpy as np
+import yaml
+
+here = pathlib.Path(__file__).parent
+# Expose the default template
+with open(here / "templatePDF.info", "r") as o:
+    template_info = yaml.safe_load(o)
+with open(here / "Toy.info", "r") as t:
+    Toy_info = yaml.safe_load(t)
+
+
+def load_info_from_file(pdfset_name):
+    """
+    Load the info file from a parent pdf.
+
+    Parameters
+    ----------
+        pdfset_name : str
+            parent pdf name
+
+    Returns
+    -------
+        dict :
+            info dictionary
+    """
+    src = pathlib.Path(lhapdf.paths()[0]) / pdfset_name
+    with open(src / ("%s.info" % (pdfset_name)), "r") as o:
+        info = yaml.safe_load(o)
+    return info
+
+
+def load_blocks_from_file(pdfset_name, member):
+    """
+    Load a pdf from a parent pdf.
+
+    Parameters
+    ----------
+        pdfset_name : str
+            parent pdf name
+        member : int
+            pdf member
+
+    Returns
+    -------
+        list(dict) :
+            pdf blocks of data
+
+    """
+    pdf = lhapdf.mkPDF(pdfset_name, member)
+    src = pathlib.Path(lhapdf.paths()[0]) / pdfset_name
+    # read actual file
+    cnt = []
+    with open(src / ("%s_%04d.dat" % (pdfset_name, pdf.memberID)), "r") as o:
+        cnt = o.readlines()
+    # file head
+    head_section = cnt.index("---\n")
+    blocks = []
+    while head_section < len(cnt) - 1:
+        # section head
+        next_head_section = cnt.index("---\n", head_section + 1)
+        # determine participating pids
+        xgrid = np.array(cnt[head_section + 1].strip().split(" "), dtype=np.float_)
+        Q2grid = np.array(cnt[head_section + 2].strip().split(" "), dtype=np.float_)
+        pids = np.array(cnt[head_section + 3].strip().split(" "), dtype=np.int_)
+        # data
+        data = []
+        for l in cnt[head_section + 4 : next_head_section]:
+            elems = re.split(r"\s+", l.strip())
+            elems = np.array(elems, dtype=np.float_)
+            data.append(elems)
+        blocks.append(dict(xgrid=xgrid, Q2grid=Q2grid, pids=pids, data=np.array(data)))
+        # iterate
+        head_section = next_head_section
+    return blocks
+
+
+def load_head_from_file(pdfset_name, member):
```

Comment:
Why do we need this function? keep in mind that reading files is an "expensive" task ... and the files are already beeing opened in `load_blocks_from_file` - then why not read it there? keep in mind that in Python you can return a tuple:
```python
def f(): return 1,2
a,b = f()
```

Reply:
Yes, done. Now there is also the possibility to overwrite every single head of every member file. Don't know if this will be useful in future but I think providing much flexibility is a good thing :)